packages/backend/server/src/core/workspaces/resolvers/workspace.ts

```diff
@@ -650,6 +620,40 @@ export class WorkspaceResolver {
     return true;
   }
 
+  private async acceptInviteByLink(user: CurrentUser, workspaceId: string) {
+    const seatAvailable = await this.quota.tryCheckSeat(workspaceId);
+    if (seatAvailable) {
+      const role = await this.models.workspaceUser.set(
+        workspaceId,
+        user.id,
+        WorkspaceRole.Collaborator,
+        WorkspaceMemberStatus.UnderReview
+      );
+      await this.workspaceService.sendReviewRequestNotification(role.id);
+      return;
+    }
+
+    const isTeam = await this.workspaceService.isTeamWorkspace(workspaceId);
+    // only team workspace allow over limit
+    if (isTeam) {
+      const role = await this.models.workspaceUser.set(
+        workspaceId,
+        user.id,
+        WorkspaceRole.Collaborator,
+        WorkspaceMemberStatus.NeedMoreSeatAndReview
+      );
+      await this.workspaceService.sendReviewRequestNotification(role.id);
```

Comment:
Keep the notification logic consistent with `seatAvailable`, both require owner and admin review